pkg/alertmanager/multitenant.go

```diff
@@ -825,13 +825,22 @@ func (am *MultitenantAlertmanager) setConfig(cfg alertspb.AlertConfigDesc) error
 	var userAmConfig *amconfig.Config
 	var err error
 	var hasTemplateChanges bool
+	var userTemplateDir = filepath.Join(am.getTenantDirectory(cfg.User), templatesDir)
+	var filesToRemove = make(map[string]string)
+
+	if oldTemplateFiles, err := ioutil.ReadDir(userTemplateDir); err == nil {
```

Comment:
I think this deserves a comment like "first we list all the files, then we will remove those from the list that are in use, then we will remove the files that are never used".

Reply:
>  Is there any danger that there could be a file there for some other reason?


I dont think so. Cortex expose an api to configure alert manager for a tenant and [this](https://cortexmetrics.io/docs/api/#set-alertmanager-configuration) API receives the template files and the prometheus [AlertManager configuration](https://prometheus.io/docs/alerting/latest/configuration/) on the same call. On a regular prometheus alert manager, the config file are stored in the filesystem and referenced by the configuration. Cortex implementation receives both on the same call, drop the template files on the filesystem and reference them on the alert manager config config.

> Could someone remove a template, then decide to use it again but you deleted it?

As we configure the templates and the alert manager on the same request, we should never have a alert manager configuration that uses a template that is not defined in the request.

> I think this deserves a comment like "first we list all the files, then we will remove those from the list that are in use, then we will remove the files that are never used".

Make sense.

Reply:
Added the comments